api/python/tests/test_bucket.py

```diff
@@ -246,3 +247,25 @@ def test_bucket_config(self, config_mock, bucket_config_mock):
         b.config('https://bar.foo/config.json')
         assert not config_mock.called
         bucket_config_mock.assert_called_once_with('test-bucket', 'https://bar.foo/config.json')
+
+    # further testing in test_search.py
+    @patch('t4.bucket.search')
+    def test_search_no_config(self, search_mock):
+        content = {
+            'federations': ['/federation.json'],
+        }
+        federations = {
+            "buckets": [
+                {
+                    "name": "t4-testing-fake",
+                    "searchEndpoint": "https://es-fake.endpoint",
+                    "region": "us-meow"
+                },
+            ]
+        }
+        self.requests_mock.add(responses.GET, 'https://foo.bar/config.json', json=content, status=200)
+        self.requests_mock.add(responses.GET, 'https://foo.bar/federation.json', json=federations, status=200)
+        b = Bucket('s3://t4-testing-fake')
+        b.search('blah', limit=1)
```

Comment:
can you add a metadata search test?

Reply:
I missed this -- but, can you be more specific about what you want here?

There are two tests, one `test_bucket.test_search_bucket` that I wrote, and another in `test_search.test_search_bucket` that it appears Calvin wrote.  I think expanding my test would duplicate his work, and mine only covers a small amount his didn't.

If you mean specific metadata search vs plaintext, broad search, that's just a difference in the query string -- which in either case here, isn't processed anyways, because it's mocked by the time it leaves our code.

Let me know if I've missed the ball on what you want here.